pyrefly/lib/state/semantic_tokens.rs

```diff
@@ -247,6 +262,40 @@ impl SemanticTokenBuilder {
 
     fn process_stmt(&mut self, x: &Stmt) {
         match x {
+            Stmt::ClassDef(class_def) => {
+                if self.is_disabled(class_def.range) {
+                    self.push_if_in_range(
+                        class_def.name.range,
+                        SemanticTokenType::CLASS,
+                        Vec::new(),
+                    );
+                }
+                x.recurse(&mut |x| self.process_stmt(x));
+            }
+            Stmt::FunctionDef(function_def) => {
+                if self.is_disabled(function_def.range) {
+                    self.push_if_in_range(
+                        function_def.name.range,
+                        SemanticTokenType::FUNCTION,
+                        Vec::new(),
+                    );
+                }
+                x.recurse(&mut |x| self.process_stmt(x));
+            }
+            Stmt::Assign(assign) => {
+                if self.is_disabled(assign.range()) {
+                    for target in &assign.targets {
+                        if let Expr::Name(name) = target {
+                            self.push_if_in_range(
+                                name.range,
+                                SemanticTokenType::VARIABLE,
+                                Vec::new(),
+                            );
+                        }
+                    }
+                }
+                x.recurse(&mut |x| self.process_stmt(x));
+            }
```

Comment:
@asukaminato0721 Would you mind clarifying why this block of code was added? Trying to retroactively understand the logic here.

Reply:
- remove semantic tokens for unreachable code
- [apparently based pyright does it](https://github.com/facebook/pyrefly/pull/1391#issuecomment-3460153362)
- there's no deprecated semantic token so this is what was done

I am not sure how useful it is to remove semantic tokens. I think the useful part is the diagnostic.

Reply:
I'm still confused why this would "remove" semantic tokens -- quite the contrary we seems to be adding semantic tokens when these AST nodes falls into disabled ranges.

Is the condition here supposed to be `if !self.is_disabled(...)` (instead of `if self.is_disabled(...)`)? Even then I don't think it'll work as intended given that the binding-based scan would always push these names. If this functionality is not important perhaps we could just remove this part of the logic?